plugin/storage/es/factory.go

```diff
@@ -317,6 +317,30 @@ func (f *Factory) CreateSamplingStore(maxBuckets int) (samplingstore.Store, erro
 		Lookback:               f.primaryConfig.AdaptiveSamplingLookback,
 		MaxDocCount:            f.primaryConfig.MaxDocCount,
 	})
+
+	if f.primaryConfig.CreateIndexTemplates && !f.primaryConfig.UseILM {
+		mappingBuilder := mappings.MappingBuilder{
```

Comment:
this is largely identical to L269, please refactor into a helper function that accepts a `cfg`

Reply:
I added following helper function. https://github.com/JaeguKim/jaeger/blob/aae3616be29bcb6d2e1f7521df5e545402078825/plugin/storage/es/factory.go#L328-L340